code/game/machinery/floodlight.dm

```diff
@@ -30,15 +30,8 @@
 	icon_state = "flood[open ? "o" : ""][open && cell ? "b" : ""]0[on]"
 
 /obj/machinery/floodlight/process()
-	if(!cell && on)
-		on = FALSE
-		visible_message("<span class='warning'>[src] shuts down due to lack of power!</span>")
-		update_icon(UPDATE_ICON_STATE)
-		set_light(0)
-		return
 	if(on)
-		cell.charge -= use
-		if(cell.charge <= 0)
+		if(!cell.use(use))
```

Comment:
Will there be a situation where this processes without a cell? If so, this will runtime